SingularityService/src/main/java/com/hubspot/singularity/hooks/LoadBalancerClientImpl.java

```diff
@@ -130,6 +130,14 @@ private BaragonResponse readResponse(Response response) {
   private SingularityLoadBalancerUpdate sendRequestWrapper(LoadBalancerRequestId loadBalancerRequestId, LoadBalancerMethod method, Request request, BaragonRequestState onFailure) {
     final long start = System.currentTimeMillis();
     final LoadBalancerUpdateHolder result = sendRequest(loadBalancerRequestId, request, onFailure);
+
+    if ((method != LoadBalancerMethod.CHECK_STATE && method != LoadBalancerMethod.PRE_ENQUEUE) &&
+        result.state == BaragonRequestState.FAILED
+        && result.message.orElse("").contains("is already enqueued with different parameters")) {
+      LOG.info("Baragon request {} already in the queue, will fetch current state instead", loadBalancerRequestId.getId());
```

Comment:
Bit of a nit, but is there any cleaner way to check for this state other than matching the against the message string? Or if we do have to do a string match, can we put the string into a constant somewhere?

Reply:
Yeah, the longer term way would be to make baragon return a different status code. Right now this is the only thing that differentiates it from other non-specific failures. Can definitely put it in a constant for the time being